internal/controller/controllers/local_cluster_import_controller.go

```diff
@@ -0,0 +1,532 @@
+			return err
+		}
+		secret.Data = data
+		return nil
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.client, &secret, mutateFn)
+	if err != nil {
+		return errors.Wrapf(err, "could not create or secret %s in namespace %s", name, namespace)
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) createOrUpdateAgentClusterInstall(ctx context.Context, numberOfControlPlaneNodes int, proxy *configv1.Proxy, instance *aiv1beta1.AgentServiceConfig) error {
+	agentClusterInstall := hiveext.AgentClusterInstall{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.localClusterName,
+			Namespace: r.localClusterName,
+		},
+	}
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, &agentClusterInstall, r.client.Scheme()); err != nil {
+			return err
+		}
+		userManagedNetworkingActive := true
+		agentClusterInstall.Spec.Networking.UserManagedNetworking = &userManagedNetworkingActive
+		agentClusterInstall.Spec.ClusterDeploymentRef = v1.LocalObjectReference{
+			Name: r.localClusterName,
+		}
+		agentClusterInstall.Spec.ImageSetRef = &hivev1.ClusterImageSetReference{
+			Name: "local-cluster-image-set",
+		}
+		agentClusterInstall.Spec.ProvisionRequirements = hiveext.ProvisionRequirements{
+			ControlPlaneAgents: numberOfControlPlaneNodes,
+		}
+		if proxy != nil {
+			agentClusterInstall.Spec.Proxy = &hiveext.Proxy{
+				HTTPProxy:  proxy.Spec.HTTPProxy,
+				HTTPSProxy: proxy.Spec.HTTPSProxy,
+				NoProxy:    proxy.Spec.NoProxy,
+			}
+		}
+		return nil
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.client, &agentClusterInstall, mutateFn)
+	if err != nil {
+		return errors.Wrap(err, "could not create or AgentClusterInstall")
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) createOrUpdateClusterDeployment(ctx context.Context, pullSecret *v1.Secret, dns *configv1.DNS, kubeConfigSecret *v1.Secret, instance *aiv1beta1.AgentServiceConfig) error {
+	if pullSecret == nil {
+		r.log.Errorf("Skipping creation of cluster deployment due to nil pullsecret")
+		return nil
+	}
+	if dns == nil {
+		r.log.Errorf("Skipping creation of cluster deployment due to nil dns")
+		return nil
+	}
+	if kubeConfigSecret == nil {
+		r.log.Errorf("Skipping creation of cluster deployment due to nil kubeconfigsecret")
+		return nil
+	}
+	if instance == nil {
+		r.log.Errorf("Skipping creation of cluster deployment due to nil agentServiceConfig")
+		return nil
+	}
+	clusterDeployment := hivev1.ClusterDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.localClusterName,
+			Namespace: r.localClusterName,
+		},
+	}
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, &clusterDeployment, r.client.Scheme()); err != nil {
+			return err
+		}
+		clusterDeployment.Spec.Installed = true
+		clusterDeployment.Spec.ClusterMetadata = &hivev1.ClusterMetadata{
+			ClusterID:                "",
+			InfraID:                  "",
```

Comment:
Cluster id is not used for day2 
infra id - doesn't really exist and we cannot create a default one because the cluster may require static network or other overrides that we don't have control over, so creating something that will not work can confuse the user.